### ISSUE TITLE
[Feature request] Create new diag tab with some useful features

### ISSUE THREAD

**Issue by token0** [microg/android_packages_apps_GmsCore#1070]
#

You even could make it active by tapping 7 times on microG version (to make it less dangerous and less clutter).

#### 

The Functions:

1. Check which location provider packages are currently used.  (is this even possible without xposed?)

    * `com.android.location.fused` — android's default flp
    * `com.google.android.gms` — microg/unifiednlp
    * `com.qualcomm.location` — qualcomm's location service with IZat, known for leaking user's location and IP — should be disabled for correct working unifiednlp! (pm disable com.qualcomm.location)

    also these can be configured in framework's strings.xml (package names) and bools.xml (enable or disable location package overlay), but we could have more convenient way to at least check these values.

    * could be anything for other vendors…

#### 

2. Button(s) to toggle (enable|disable) correct permissions, such as

    * `pm grant com.android.vending android.permission.FAKE_PACKAGE_SIGNATURE`

    * `pm grant com.google.android.gms android.permission.ACCESS_FINE_LOCATION`
    * `pm grant com.google.android.gms android.permission.ACCESS_COARSE_LOCATION`

    * `pm revoke com.qualcomm.location android.permission.ACCESS_FINE_LOCATION`
    * `pm revoke com.qualcomm.location android.permission.ACCESS_COARSE_LOCATION`

    * `pm disable com.qualcomm.location`
    * `pm enable com.qualcomm.location`

    * etc…

**Comment by breversa** [comment]
Good idea ! I thought of something similar/in addition to your idea in https://github.com/microg/android_packages_apps_GmsCore/issues/1058
